ntoskrnl/mm/ARM3/virtual.c

```diff
@@ -5176,6 +5175,10 @@ NtAllocateVirtualMemory(IN HANDLE ProcessHandle,
     }
 
 FailPathNoLock:
+    if (!NT_SUCCESS(Status))
```

Comment:
I don't generally like the idea of doing that release at a completely unrelated place than where the quota was originally charged (observe how all that code directly above there corresponds to a different execution branch than the one taken by the code block where the quota was charged...)
Cc @ThFabba @tkreuzer for ideas.